app/controllers/concerns/decidim/participatory_documents/needs_pdf_document.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module Decidim
+  module ParticipatoryDocuments
+    # This module provides methods and helpers for the pdf_viewer/editor
+    module NeedsPdfDocument
+      extend ActiveSupport::Concern
+
+      included do
+        helper_method :document, :box_color_as_rgba, :pdf_custom_style, :pdf_i18n
+
+        protected
+
+        def document
+          @document ||= Decidim::ParticipatoryDocuments::Document.find_by(component: current_component)
+        end
+
+        def box_color_as_rgba(document, opacity: nil)
+          opacity ||= document.box_opacity
+          return "rgba(30, 152, 215, 0.12);" unless document.box_color.present? && opacity.present?
+
+          document.box_color + (opacity * 2.55).round.to_s(16).rjust(2, "0")
+        end
+
+        def suggestions_modal_color_as_hsla(document, opacity: 0.95)
```

Comment:
```suggestion
        def box_color_as_hsla(document, opacity: 0.95)
```
In this case I'd prefer to maintain the same name for all the color processors. This is because this can be used in any context (maybe one day we will use half for the boxes)